packages/app-project/src/screens/SubjectTalkPage/components/SubjectTalkData/components/Tags/TagsContainer.jsx

```diff
@@ -66,204 +66,271 @@ function TagsContainer({
                 }
               }
               return t
             })
-            return newData
           }
         },
-        { revalidate: false }
+        { revalidate: false } // don't need to revalidate votableTags because optimistic data is enough
       )
     }
 
-    // Remove the tag vote from the user's tag votes
     mutateTagVotes(
-      prevData => {
+      async (prevData) => {
+        setVoteUpdating(true)
+        // remove the "tag vote" from the database and the database changes the relevant votableTag's vote_count to 0
+        await removeTagVote(tagVoteToRemove.id)
+        setVoteUpdating(false)
+
+        // this async function must return something to put into the tagVotes data array
+        // so remove the tag vote from the tagVotes array
         if (!prevData) {
           return []
         } else {
-          const newData = prevData.filter(vote => vote.id !== tagVoteToRemove.id)
-          return newData
+          return prevData.filter(
+            vote => vote.id !== tagVoteToRemove.id
+          )
         }
       },
-      { revalidate: false }
+      {
+        // optimistically remove the tag vote from the tagVotes array before the network request happens
+        optimisticData: prevData => {
+          if (!prevData) {
+            return []
+          } else {
+            return prevData.filter(
+              vote => vote.id !== tagVoteToRemove.id
+            )
+          }
+        },
+        // don't need to revalidate tagVotes because optimisticData is enough
+        revalidate: false,
+        // rollbackOnError will respond to an error thrown by removeTagVote()
+        // tagVotes will be rolled back, but the above mutations to popularTags or votableTags will not,
+        // this could be improved to rollback all mutations in handleRemoveVote()
+        rollbackOnError(err) {
+          console.error(err)
+          return true
+        }
+      }
     )
-
-    try {
-      removeTagVote(tagVoteToRemove.id)
-    } catch (error) {
-      console.error(error)
-    }
   }
 
   function handleClick(tag) {
     if (!tag.vote_count) {
-      handleCreateVotableTag(tag)
+      handleCreateVotableTag(tag) // add vote to popular tag
     } else if (tag.userVoted) {
-      handleRemoveVote(tag)
+      handleRemoveVote(tag) // remove vote is the same process regardless of tag type
     } else {
-      handleAddVote(tag)
+      handleAddVote(tag) // add vote to votable tag
     }
   }
 
   return (
     <Tags
-      loading={tagsIsLoading || votableTagsIsLoading || tagVotesIsLoading}
-      error={tagsError || votableTagsError || tagVotesError}
+      loading={popularTagsIsLoading || votableTagsIsLoading || tagVotesIsLoading}
+      error={popularTagsError || votableTagsError || tagVotesError}
       tags={combinedTags}
       onTagClick={handleClick}
       userId={userId}
-      voteUpdating={voteUpdating}
+      voteUpdating={voteUpdating || tagVotesIsValidating}
```

Comment:
Use the developer tools network throttling to 3G for testing when `voteUpdating` is true in Tags.